mapper-javabean/src/main/java/org/hibernate/search/v6poc/entity/javabean/JavaBeanMappingInitiator.java

```diff
@@ -9,38 +9,33 @@
-import org.hibernate.search.v6poc.entity.javabean.mapping.impl.JavaBeanMappingFactory;
-import org.hibernate.search.v6poc.entity.javabean.mapping.impl.JavaBeanMappingKey;
+import org.hibernate.search.v6poc.entity.javabean.impl.JavaBeanMappingInitiatorImpl;
 import org.hibernate.search.v6poc.entity.javabean.model.impl.JavaBeanBootstrapIntrospector;
-import org.hibernate.search.v6poc.entity.pojo.mapping.spi.PojoMappingInitiatorImpl;
+import org.hibernate.search.v6poc.entity.pojo.mapping.PojoMappingInitiator;
 
-public final class JavaBeanMappingInitiator extends PojoMappingInitiatorImpl<JavaBeanMapping> {
+public interface JavaBeanMappingInitiator extends PojoMappingInitiator<JavaBeanMapping> {
 
-	public JavaBeanMappingInitiator(SearchMappingRepositoryBuilder mappingRepositoryBuilder) {
-		this( mappingRepositoryBuilder, true );
+	static JavaBeanMappingInitiator create(SearchMappingRepositoryBuilder mappingRepositoryBuilder) {
+		return create( mappingRepositoryBuilder, true );
 	}
 
-	public JavaBeanMappingInitiator(SearchMappingRepositoryBuilder mappingRepositoryBuilder,
+	static JavaBeanMappingInitiator create(SearchMappingRepositoryBuilder mappingRepositoryBuilder,
 			boolean annotatedTypeDiscoveryEnabled) {
-		this( mappingRepositoryBuilder, MethodHandles.publicLookup(), annotatedTypeDiscoveryEnabled, false );
+		return create( mappingRepositoryBuilder, MethodHandles.publicLookup(), annotatedTypeDiscoveryEnabled, false );
 	}
 
-	public JavaBeanMappingInitiator(SearchMappingRepositoryBuilder mappingRepositoryBuilder,
+	static JavaBeanMappingInitiator create(SearchMappingRepositoryBuilder mappingRepositoryBuilder,
 			boolean annotatedTypeDiscoveryEnabled, boolean multiTenancyEnabled) {
-		this( mappingRepositoryBuilder, MethodHandles.publicLookup(), annotatedTypeDiscoveryEnabled, multiTenancyEnabled );
+		return create(
+				mappingRepositoryBuilder, MethodHandles.publicLookup(),
+				annotatedTypeDiscoveryEnabled, multiTenancyEnabled
+		);
 	}
 
-	public JavaBeanMappingInitiator(SearchMappingRepositoryBuilder mappingRepositoryBuilder,
+	static JavaBeanMappingInitiator create(SearchMappingRepositoryBuilder mappingRepositoryBuilder,
 			MethodHandles.Lookup lookup, boolean annotatedTypeDiscoveryEnabled, boolean multiTenancyEnabled) {
-		this( mappingRepositoryBuilder, new JavaBeanBootstrapIntrospector( lookup ), annotatedTypeDiscoveryEnabled, multiTenancyEnabled );
-	}
-
-	private JavaBeanMappingInitiator(SearchMappingRepositoryBuilder mappingRepositoryBuilder,
-			JavaBeanBootstrapIntrospector introspector, boolean annotatedTypeDiscoveryEnabled, boolean multiTenancyEnabled) {
-		super(
-				mappingRepositoryBuilder, new JavaBeanMappingKey(),
-				new JavaBeanMappingFactory(),
-				introspector, false,
+		return new JavaBeanMappingInitiatorImpl(
+				mappingRepositoryBuilder, new JavaBeanBootstrapIntrospector( lookup ),
 				annotatedTypeDiscoveryEnabled, multiTenancyEnabled
```

Comment:
Is it really such a good idea to have these static methods in an API?

Reply:
We have to offer an entry point to our API at some point, and this entry point cannot be an instance method, since users don't have any instance to call it on. So the entry point must be either a constructor or a static method.

These are the only differences I can see between a constructor and a static factory method:

* we can allow people to extend a class that exposes a public constructor, whereas classes instantiated through a static factory method cannot be easily extended.
* we can introduce logic to choose the type to instantiate in a static factory method, whereas the type is pre-determined in a constructor.

Other than that, they are practically equivalent.

I don't expect people to extend this class (if people have to extend something, I'd rather they use an SPI), so static methods are as good as constructors here in my opinion.

If you're really worried about these methods, I can make `JavaBeanMappingInitiator` a class that delegates to `JavaBeanMappingInitiatorImpl`. Then people will use the constructor of this class instead of static methods. This will require a bit of class renaming though.
Or I can introduce a  `JavaBeanMappingInitiatorFactory` class that will be instantiated through its constructor or a singleton accessor (`public static Factory get()`), then will offer non-static `create` methods.
Or, if adding more parameters is what worries you, I can use a builder pattern (`JavaBeanMappingInitiator.builder( mappingRpositoryBuilder ).enableMultiTenancy().build()`).

If I really had to pick one, I'd go with the builder solution. But I don't really feel it's necessary at the moment, and it could be added later, next to the current static methods.

Reply:
OK, let's keep it that way then.